src/opera_disp_tms/__main__.py

```diff
@@ -20,53 +18,36 @@ def __call__(self, parser, namespace, values, option_string=None):
         setattr(namespace, self.dest, value)
 
 
-class Bbox(argparse.Action):
+class Frames(argparse.Action):
     def __call__(self, parser, namespace, values, option_string=None):
         try:
-            bbox = tuple(int(item) for sublist in values for item in sublist)
+            frames = [int(item) for sublist in values for item in sublist]
         except ValueError:
-            parser.error(f'{self.dest} values must be integers ordered [min lon, min lat, max lon, max lat]')
-        if len(bbox) != 4:
-            parser.error(f'{self.dest} must have exactly 4 values ordered [min lon, min lat, max lon, max lat]')
-        if not (-180 <= bbox[0] <= bbox[2] <= 180 and -90 <= bbox[1] <= bbox[3] <= 90):
-            parser.error(f'{self.dest} must be ordered [min lon, min lat, max lon, max lat]')
-        setattr(namespace, self.dest, bbox)
-
-
-def generate_mosaic_geotiff(
-    tile_type: str, bbox: tuple[int, int, int, int], direction: str, begin_date: datetime, end_date: datetime
-) -> Path:
-    metadata_geotiff = create_tile_for_bbox(bbox, direction=direction)
-    if not metadata_geotiff:
-        raise ValueError(f'No data found for bounding box {bbox} and direction {direction}')
-
-    if tile_type == 'displacement':
-        mosaic_geotiff = create_sw_disp_tile(metadata_geotiff, begin_date, end_date)
-    elif tile_type == 'secant_velocity':
-        mosaic_geotiff = create_sw_vel_tile(metadata_geotiff, begin_date, end_date, secant=True)
-    else:
-        raise ValueError(f'Unsupported tile type: {tile_type}')
-
-    return mosaic_geotiff
+            parser.error(f'{self.dest} values must be integers between 1 and 46986')
+        for frame in frames:
+            if not (1 <= frame <= 46986):
```

Comment:
Might be worth checking if `frame` is an int, or casting it to an int

Reply:
Frames are casted to ints on line 24